WireAPI/Sources/WireAPI/APIs/ConversationsAPI/ConversationsAPIV5.swift

```diff
@@ -20,4 +20,151 @@ import Foundation
 
 class ConversationsAPIV5: ConversationsAPIV4 {
     override var apiVersion: APIVersion { .v5 }
+
+    override func getConversations(for identifiers: [QualifiedID]) async throws -> ConversationList {
+        let parameters = GetConversationsParametersV0(qualifiedIdentifiers: identifiers)
+        let body = try JSONEncoder.defaultEncoder.encode(parameters)
+        let resourcePath = "\(pathPrefix)/conversations/list"
+
+        let request = HTTPRequest(
+            path: resourcePath,
+            method: .post,
+            body: body
+        )
+        let response = try await self.httpClient.executeRequest(request)
+
+        // Remove in v5: remove handling of error code 400
+        return try ResponseParser()
+            .success(code: 200, type: QualifiedConversationListV5.self) // Change in v5
+            .parse(response)
+    }
+}
+
+// MARK: Decodables
+
+private struct QualifiedConversationListV5: Decodable, ToAPIModelConvertible {
```

Comment:
As I understand, the success payload didn't change at all right? If that's the case, couldn't we keep using the previous version?

Reply:
The issue I faced here is that each of these `QualifiedConversationList*` objects refers to a conversation like `let found: [ConversationV5]`. So V4 would use `ConversationV4`, which doesn't contain `cipherSuite` and `epochTimestamp`.

I've tried to make a generic version of `QualifiedConversationList<Conversation: Decodable & ToAPIModelConvertible>`, but as far as I got I didn't make the compiler understand that the generic is always transformable into `found: [Conversation]` for the final `public ConversationList`.

Do you know another approach in mind?